src/dotnet/commands/dotnet-run3/Run3Command.cs

```diff
@@ -87,30 +75,25 @@ private ITaskItem GetRunInformation()
 
             ProjectInstance projectInstance = new ProjectInstance(Project, globalProperties, null);
 
-            BuildRequestData buildRequestData = new BuildRequestData(projectInstance, new string[] { GetRunInformationTaskName });
-            BuildParameters buildParameters = new BuildParameters();
-
-            BuildResult result = BuildManager.DefaultBuildManager.Build(buildParameters, buildRequestData);
-
-            TargetResult runInfoResult;
-            if (!result.ResultsByTarget.TryGetValue(GetRunInformationTaskName, out runInfoResult))
+            string runProgram = projectInstance.GetPropertyValue(RunCommandPropName);
+            if (string.IsNullOrEmpty(runProgram))
             {
-                throw new InvalidOperationException($"Could not find a target named '{GetRunInformationTaskName}' in your project. Please ensure 'dotnet run' supports this project.");
+                throw new InvalidOperationException($"The property named '{RunCommandPropName}' does not have a value in your project. Please ensure 'dotnet run' supports this project.");
             }
 
-            if (runInfoResult.ResultCode != TargetResultCode.Success)
-            {
-                throw new InvalidOperationException($"Could not get the run information for project {Project}. An internal MSBuild error has occured" + Environment.NewLine +
-                    runInfoResult.Exception?.ToString());
-            }
+            string runArguments = projectInstance.GetPropertyValue(RunArgumentsPropName);
+            string runWorkingDirectory = projectInstance.GetPropertyValue(RunWorkingDirectoryPropName);
 
-            ITaskItem runInfoItem = runInfoResult.Items.FirstOrDefault(i => i.ItemSpec == projectInstance.FullPath);
-            if (runInfoItem == null)
+            string fullArguments = runArguments;
```

Comment:
I think this is going to give us some trouble with escaping if it's widely used.

We'll be requiring users to know about how escaping works to do this right, things like you can't have a trailing backslash in your path if it's surrounded by quotes are non-obvious and cause user confusion frequently. 

What if the run arguments were a set of items that we could pass through the argument escaper?